core/src/core/pmmr/pmmr.rs

```diff
@@ -665,14 +668,27 @@ pub fn family_branch(pos: u64, last_pos: u64) -> Vec<(u64, u64)> {
 }
 
 /// Gets the position of the rightmost node (i.e. leaf) beneath the provided subtree root.
-pub fn bintree_rightmost(num: u64) -> u64 {
-	num - bintree_postorder_height(num)
+pub fn bintree_rightmost(pos: u64) -> u64 {
+	pos - bintree_postorder_height(pos)
 }
 
 /// Gets the position of the rightmost node (i.e. leaf) beneath the provided subtree root.
-pub fn bintree_leftmost(num: u64) -> u64 {
-	let height = bintree_postorder_height(num);
-	num + 2 - (2 << height)
+pub fn bintree_leftmost(pos: u64) -> u64 {
+	let height = bintree_postorder_height(pos);
+	pos + 2 - (2 << height)
+}
+
+/// Iterator over all leaf pos beneath the provided subtree root (including the root itself).
+pub fn bintree_leaf_pos_iter(pos: u64) -> impl Iterator<Item = u64> {
+	let leaf_start = max(1, bintree_leftmost(pos as u64));
+	let leaf_end = bintree_rightmost(pos as u64);
+	(leaf_start..=leaf_end).filter(|x| is_leaf(*x))
+}
+
+/// Iterator over all pos beneath the provided subtree root (including the root itself).
+pub fn bintree_pos_iter(pos: u64) -> impl Iterator<Item = u64> {
+	let leaf_start = max(1, bintree_leftmost(pos as u64));
```

Comment:
why the max(1, ?

Reply:
it's redundant in theory however the tests are constructed with the assumption that `bintree_pos_iter(0)` returns an empty iterator which is produced by this, so I don't want to break this assumption ATM.